WebXR: Content is not loading in Private Browsing

grorg opened the pull request:
#### a6ab20ed7b64f6163f87f90d10397ef5bd6c553a
<pre>
WebXR: Content is not loading in Private Browsing
<a href="https://bugs.webkit.org/show_bug.cgi?id=259462">https://bugs.webkit.org/show_bug.cgi?id=259462</a>
rdar://108818635

Reviewed by Tim Horton and Mike Wyrzykowski.

The sandbox on iOS Private Browsing blocks WebXR from accessing its
cache directory on visionOS. Specifically add those directories
to the sandbox extension.

* Source/WebKit/Shared/WebProcessCreationParameters.h:
* Source/WebKit/Shared/WebProcessCreationParameters.serialization.in:
* Source/WebKit/UIProcess/Cocoa/WebProcessPoolCocoa.mm:
* Source/WebKit/WebProcess/cocoa/WebProcessCocoa.mm:

Canonical link: <a href="https://commits.webkit.org/266346@main">https://commits.webkit.org/266346@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/3e9642bdb9d70de1c3cac77eec2efde4ed5d1251

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/13591 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/13905 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/14238 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/15327 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/12910 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/16413 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/13985 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/15602 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/13757 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/14401 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/11510 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/16026 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/11689 "Passed tests") | | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/19298 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/12764 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/12433 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/15635 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/12942 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/10833 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/12212 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/3309 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/16542 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/12785 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

hortont424 commented on Source/WebKit/UIProcess/WebsiteData/Cocoa/WebsiteDataStoreCocoa.mm:
```diff
@@ -927,9 +927,12 @@ static NavigatingToAppBoundDomain schemeOrDomainIsAppBound(const String& host, c
         if (!isPersistent())
             m_resolvedContainerCachesWebContentDirectory = emptyString();
-        else {
+        else
```
I'm confused. Doesn't this mean that ephemeral sessions will leak caches to disk? Why is that OK?

mwyrzykowski replied:
This is a good point. Dismissing my approval for now. Agreed allowing WebXR to write to the cache in ephemeral sessions seems questionable.

grorg replied:
Good point. The other option was to allow the sandbox to read the cache in the normal location, which also might be bad. I'll ask @brentfulgham

rr-codes commented on Source/WebKit/Shared/WebProcessCreationParameters.h:
```diff
@@ -200,6 +200,11 @@ struct WebProcessCreationParameters {
 #endif
 
+#if PLATFORM(VISION)
+    // FIXME(??): Remove when GPU Process is fully enabled
```
should the ?? be here?

hortont424 replied:
“??”??

hortont424 reviewed (approved):
Probably want Per Arne/Brent/someone review

grorg committed:
WebXR: Content is not loading in Private Browsing
https://bugs.webkit.org/show_bug.cgi?id=259462
rdar://108818635

Reviewed by Tim Horton and Mike Wyrzykowski.

The sandbox on iOS Private Browsing blocks WebXR from accessing its
cache directory on visionOS. Specifically add those directories
to the sandbox extension.

* Source/WebKit/Shared/WebProcessCreationParameters.h:
* Source/WebKit/Shared/WebProcessCreationParameters.serialization.in:
* Source/WebKit/UIProcess/Cocoa/WebProcessPoolCocoa.mm:
* Source/WebKit/WebProcess/cocoa/WebProcessCocoa.mm:

Canonical link: https://commits.webkit.org/266346@main
